sqlglot/tokens.py

```diff
@@ -566,17 +566,22 @@ class Tokenizer(metaclass=_Tokenizer):
 
     def __init__(
         self,
-        identifier=None,
+        identifiers=None,
         escape=None,
     ):
         """
         Tokenizer consumes a sql string and produces an array of :class:`~sqlglot.tokens.Token`
 
         Args
-            identifier (str): the identifier character
+            identifier (array): identifier characters and/or tuples when different start and end identifiers
             escape (str): the escape code character
         """
-        self.identifier = identifier or '"'
+        self.identifiers = dict(
```

Comment:
sorry again i'm being a performance nut, but this looks expensive, maybe let's make the api identifier only take in an array of tuples, and then have this tuple stuff happen in the class loader

Reply:
It's a tradeoff of uglier identifiers definitions in the sql dialects vs having the extra conditional in the dict comprehension. Whichever you prefer is fine by me.